plugins/intercom/hooks.js

```diff
@@ -0,0 +1,82 @@
+import { Hooks } from "PluginSDK";
+
+import DOMUtils from "#SRC/js/utils/DOMUtils";
+
+import AuthStore from "#SRC/js/stores/AuthStore";
+import IntercomStore from "#SRC/js/stores/IntercomStore";
+import MetadataStore from "#SRC/js/stores/MetadataStore";
+
+import EventTypes from "#SRC/js/constants/EventTypes";
+
+const SDK = require("./SDK").getSDK();
+
+module.exports = {
+  actions: ["pluginsConfigured", "userLoginSuccess", "userLogoutSuccess"],
+
+  initialize(configuration) {
+    this.configuration = configuration;
+
+    this.actions.forEach(action => {
+      SDK.Hooks.addAction(action, this[action].bind(this));
+    });
+
+    this.bootIntercom = this.bootIntercom.bind(this);
+
+    const intercomScript = [
+      `var APP_ID = "${this.configuration.appId}";`,
+      `(function(){var w=window;var ic=w.Intercom;if(typeof ic==="function"){ic('reattach_activator');ic('update',intercomSettings);}else{var d=document;var i=function(){i.c(arguments)};i.q=[];i.c=function(args){i.q.push(args)};w.Intercom=i;function l(){var s=d.createElement('script');s.type='text/javascript';s.async=true;s.src='https://widget.intercom.io/widget/' + APP_ID;var x=d.getElementsByTagName('script')[0];x.parentNode.insertBefore(s,x);}if(w.attachEvent){w.attachEvent('onload',l);}else{w.addEventListener('load',l,false);}}})()`
+    ].join("");
+
+    DOMUtils.appendScript(global.document.head, intercomScript);
+  },
+
+  pluginsConfigured() {
+    if (AuthStore.isLoggedIn) {
+      this.bootIntercom();
+    }
+  },
+
+  userLoginSuccess() {
+    this.bootIntercom();
+  },
+
+  userLogoutSuccess() {
+    window.Intercom("shutdown");
+  },
+
+  bootIntercom() {
+    const boot = () => {
+      const user = AuthStore.getUser();
+
+      const attributes = {
+        app_id: this.configuration.appId,
+        "bootstrap-id": MetadataStore.bootstrapId,
+        clusterId: MetadataStore.clusterId,
+        "dcos-image-commit": MetadataStore.imageCommit,
+        dcosVersion: MetadataStore.version,
+        dcosVariant: MetadataStore.variant,
+        email: user.email,
+        name: user.name,
+        userId: user.uid
+      };
+
+      Object.keys(attributes).forEach(attribute => {
+        IntercomStore.addAttribute(attribute, attributes[attribute]);
+      });
+
+      window.Intercom("boot", IntercomStore.attributes);
+    };
+
+    if (!MetadataStore.version) {
+      MetadataStore.addChangeListener(EventTypes.DCOS_METADATA_CHANGE, boot);
+    }
+
+    if (!MetadataStore.clusterId) {
```

Comment:
If `version` is there then `clusterId` should also be there. To be on a safe side we could merge two checks into one so that we don't add too many listeners

Reply:
oh, those are different things!

Reply:
Though I wonder if the `IntercomStore` should listen to every other store and emit its own events and the plugin should be subscribed to `IntercomStore` only? Can we actually do a data layer real quick here? @Fabs

Reply:
I am not sure If I understand the use case of the DL here. Can you elaborate?

I see an opportunity to use a stream to simplify that code (according to my subjective perception of simple). The pattern I see is that we call boot() when a lot of things happen. There is also the async load of the JS code, and even a timeout. We could make a stream that has a source for all these events and transforms into the call to boot, also triggering the Hooks.doAction.

Reply:
Use DL to pump data into the plugin instead of writing another store